inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -105,19 +105,20 @@ def add(self, parameters: ApplicationAddSourceParameters) -> None:
         # Step 1: Verify GPG key URI from trusted repo list
         if parameters.gpg_key_name and parameters.gpg_key_uri:
             try:
-               url = parameters.gpg_key_uri
-               #URL slicing to remove the last segment (filename) from the URL 
-               source = url[:-(len(url.split('/')[-1]) + 1)]
-               verify_source(source=source, dispatcher_broker=self._dispatcher_broker)
+                url = parameters.gpg_key_uri
+                # URL slicing to remove the last segment (filename) from the URL
+                source = url[:-(len(url.split('/')[-1]) + 1)]
+                verify_source(source=source, dispatcher_broker=self._dispatcher_broker)
             except (DispatcherException, IndexError) as err:
-               raise SourceError(f"Source GPG key URI verification check failed: {err}")
+                raise SourceError(f"Source GPG key URI verification check failed: {err}")
             # Step 2: Add key (Optional)
             add_gpg_key(parameters.gpg_key_uri, parameters.gpg_key_name)
 
         # Step 3: Add the source
         try:
             create_file_with_contents(
-                os.path.join(UBUNTU_APT_SOURCES_LIST_D, parameters.source_list_file_name), parameters.sources
+                os.path.join(UBUNTU_APT_SOURCES_LIST_D,
+                             parameters.source_list_file_name), parameters.sources
             )
         except (IOError, OSError) as e:
             raise SourceError(f"Error adding application source list: {e}")
```

Comment:
**<h2>$\color{#FFB6C1}Enhancement: CodeRefactoring$</h2>**
**Suggestion:** Refactor the repeated code for writing to the sources list file into a separate method to avoid duplication and improve maintainability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        def _write_to_sources_list(self, sources: List[str], mode: str = 'a') -> None:
            """Writes the given sources to the Ubuntu OS sources list file.

            Args:
                sources: A list of source strings to write.
                mode: The file opening mode ('a' for append, 'w' for overwrite).
            """
            try:
                with open(UBUNTU_APT_SOURCES_LIST, mode) as file:
                    for source in sources:
                        file.write(f"{source}\n")
            except OSError as e:
                raise SourceError(f"Error writing to sources list: {e}") from e

        # Usage in add method
        self._write_to_sources_list(parameters.sources, mode='a')

        # Usage in update method
        self._write_to_sources_list(parameters.sources, mode='w')
```

Reply:
out of scope for formatting change, but good suggestion

Reply:
out of scope but good suggestion